src/Matmar10/Bundle/MoneyBundle/Service/CurrencyManager.php

```diff
@@ -99,13 +100,13 @@ protected function searchCurrency($currencyCodeOrCountryCode)
         // search first in the on-the-fly added currencies
         $currency = $this->searchAddedCurrencies($currencyCodeOrCountryCode);
         if(false !== $currency) {
-            return $currency;
+            return $this->addedCurrencies[$currencyCodeOrCountryCode] = $currency; //add to "addedCurrencies" will speed up the next search
```

Comment:
The reason is that it induces a "WTF" moment to the next person to read this code.
The version I've provided is 100% crystal clear that you are doing assignment, and then returning.
The version you've written only takes up one line, but it's not immediately clear whether you intended to do evaluation (==) or assignment.

My philosophy is go with what is absolutely crystal clear without a doubt - even at the expense of adding more lines.